hack/generator/pkg/codegen/storage/testdata/Test_StorageTypeFactory_GeneratesExpectedResults-v20211231storage.golden

```diff
@@ -0,0 +1,54 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package v20211231storage
+
+import (
+	"github.com/Azure/azure-service-operator/hack/generated/microsoft.person/v20211231storage"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+//Storage version of v20211231.Person
+type Person struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              v20211231storage.Person_Spec   `json:"spec,omitempty"`
+	Status            v20211231storage.Person_Status `json:"status,omitempty"`
+}
+
+func (person *v20211231storage.Person) OriginalGVK() *schema.GroupVersionKind {
+	return &schema.GroupVersionKind{
+		Group:   GroupVersion.Group,
+		Version: person.Spec.OriginalVersion,
+		Kind:    "Person",
+	}
+}
+
+// +kubebuilder:object:root=true
+//Storage version of v20211231.Person
+type PersonList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []v20211231storage.Person `json:"items"`
+}
+
+//Storage version of v20211231.Person_Spec
+type Person_Spec struct {
+	CustomerProgram *string `json:"customerProgram,omitempty"`
+	FamilyName      *string `json:"familyName,omitempty"`
+	FullName        *string `json:"fullName,omitempty"`
+	KnownAs         *string `json:"knownAs,omitempty"`
+	OriginalVersion string  `json:"original-version"`
```

Comment:
shouldn't this be `originalVersion` in the JSON?

Reply:
Yes, but I've removed `storage_type_factory` entirely. #nofixneeded